src/tqec/compile/specs/library/base.py

```diff
@@ -0,0 +1,485 @@
+                side of the different templates should be matched together.
+
+        Raises:
+            TQECException: if ``direction == Direction3D.Z``.
+
+        Returns:
+            two mappings, the first one for plaquette indices from
+            ``pipe_template`` to ``qubit_templates[0]`` and the second one from
+            ``pipe_template`` to ``qubit_templates[1]``.
+        """
+        tb1: TemplateBorder
+        tb2: TemplateBorder
+        match direction:
+            case Direction3D.X:
+                tb1 = TemplateBorder.LEFT
+                tb2 = TemplateBorder.RIGHT
+            case Direction3D.Y:
+                tb1 = TemplateBorder.TOP
+                tb2 = TemplateBorder.BOTTOM
+            case Direction3D.Z:
+                raise TQECException("This method cannot be used with a temporal pipe.")
+
+        return (
+            pipe_template.get_border_indices(tb1).to(
+                qubit_templates[0].get_border_indices(tb2)
+            ),
+            pipe_template.get_border_indices(tb2).to(
+                qubit_templates[1].get_border_indices(tb1)
+            ),
+        )
+
+    @staticmethod
+    def _get_spatial_cube_arm(spec: PipeSpec) -> SpatialArms:
+        """Returns the arm corresponding to the provided ``spec``.
+
+        Args:
+            spec: pipe specification to get the arm from.
+
+        Raises:
+            TQECException: if the provided ``spec`` is not a spatial pipe.
+            TQECException: if the two blocks connected to the pipe are both
+                spatial junctions, which is currently an unsupported case.
+
+        Returns:
+            the :class:`~tqec.compile.specs.enums.SpatialArms` instance
+            corresponding to the provided ``spec``. The returned flag only
+            contains one flag (i.e., it cannot be
+            ``SpatialArms.RIGHT | SpatialArms.UP``).
+        """
+        assert spec.pipe_kind.is_spatial
+        # Check that we do have a spatial junction.
+        assert any(spec.is_spatial for spec in spec.cube_specs)
+        # For the moment, two spatial junctions side by side are not supported.
+        if all(spec.is_spatial for spec in spec.cube_specs):
+            raise TQECException(
+                "Found 2 spatial junctions connected. This is not supported yet."
+            )
+        spatial_cube_is_first: bool = spec.cube_specs[0].is_spatial
+        match spatial_cube_is_first, spec.pipe_kind.direction:
+            case (True, Direction3D.X):
+                return SpatialArms.RIGHT
+            case (False, Direction3D.X):
+                return SpatialArms.LEFT
+            case (True, Direction3D.Y):
+                return SpatialArms.DOWN
+            case (False, Direction3D.Y):
+                return SpatialArms.UP
+            case _:
+                raise TQECException(
+                    "Should never happen as we are in a spatial (i.e., X/Y plane) junction."
+                )
```

Comment:
The limitation to only one spatial cube should be removed: linking 2 spatial cubes should be possible, and it is not that hard to do so. You can have a look at 46e9d64cd243f21ba1e3d49991694bfedf2d5570 in #510 to have an idea of how I removed that condition.